flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/status/SavepointInfo.java

```diff
@@ -63,7 +63,7 @@ public void setTrigger(
     }
 
     public void resetTrigger() {
-        this.triggerId = "";
+        this.triggerId = null;
         this.triggerTimestamp = 0L;
```

Comment:
Can't we use `Long` and `null`s as default for the timestamps as well (lastPeriodicSavepointTimestamp,triggerTimestamp) or it would break backward compatibility?

Reply:
Adding this would end-up in additional code like null checks, etc..., but what would be the benefit?

Reply:
The benefit is that there would not be `triggerTimestamp` in the status when there is nothing triggered.

I agree with @morhidi I will change this, originally I thought this would be a Java API breaking change but I believe Kubernetes never sends `null` values so it's not going to be a problem.

Reply:
Sounds fair.

Reply:
It'll keep already stored `0` values however, so no business logic should check `(triggerTimestamp == null)`, it'll be `0`